### PR TITLE
zammad: 5.1.4 -> 6.1.0

### PR DESCRIPTION
**Pull request by NetaliDev**
## Description of changes

This PR updates `zammad` to 6.1.0 and updates the module to work with the new version. It also adds me as a maintainer for the package and module.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NetaliDev** [comment]
@ofborg build zammad

**Review by Janik-Haag — COMMENTED** [review]
Test is failing for me :(

**Review by Janik-Haag — COMMENTED** [review]
Diff looks good, we should post in the [breaking-changes-announcement-for-unstable/](https://discourse.nixos.org/t/breaking-changes-announcement-for-unstable/17574?u=janik) and wait two or so days before merging since this release has breaking changes.
But the test still fails for me, now with this error: `error: boost::bad_format_string: format-string is ill-formed` did you experience the same?

**Comment by NetaliDev** [comment]
Hmm strange, the test works for me now and apparently for ofborg too.

But while we figure this out, I will post the announcement in the breaking changes thread.

File: nixos/modules/services/development/zammad.nix
Comment (SuperSandro2000) [inline nixos/modules/services/development/zammad.nix:248]: Why not use the redis socket for local things?

File: pkgs/applications/networking/misc/zammad/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/misc/zammad/default.nix:136]:
```suggestion
    redis-server &
```
lets make this cross friendly by adding redis to nativeBuildInputs

**Review by n0emis — APPROVED** [review]

File: pkgs/applications/networking/misc/zammad/default.nix
Comment (garbas) [inline pkgs/applications/networking/misc/zammad/default.nix:164]:
```suggestion
    maintainers = with maintainers; [ n0emis taeer netali ];
```

Would you be so kind and also remove me as a maintainer.

File: nixos/modules/services/development/zammad.nix
Comment (NetaliDev) [reply nixos/modules/services/development/zammad.nix:248]: Because Zammad wants to use TCP for redis by default, and the module just adopts this.

File: nixos/tests/zammad.nix
Comment (NetaliDev) [reply nixos/tests/zammad.nix:7]: I do not see myself in a position to remove others as maintainers if I do not have their consent.

File: pkgs/applications/networking/misc/zammad/default.nix
Comment (NetaliDev) [reply pkgs/applications/networking/misc/zammad/default.nix]: Done.

File: pkgs/applications/networking/misc/zammad/default.nix
Comment (NetaliDev) [reply pkgs/applications/networking/misc/zammad/default.nix]: Done.

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**
